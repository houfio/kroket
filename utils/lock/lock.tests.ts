import { act, renderHook } from '@testing-library/react-hooks';

import { useLock } from '.';

it('should hide overflow on body', () => {
  const { result: { current: [lock] } } = renderHook(() => useLock());

  act(() => {
    lock();
  });

  expect(document.body).toMatchSnapshot();
});

it('should preserve padding', () => {
  const { result: { current: [lock] } } = renderHook(() => useLock());

  document.body.innerHTML = `
    <main></main>
    <div data-kroket-fixed style="padding-right: 1rem"></div>
  `;

  act(() => {
    lock();
  });

  expect(document.body).toMatchSnapshot();
});

it('should restore fixed padding', () => {
  const { result: { current: [lock, unlock] } } = renderHook(() => useLock());

  document.body.innerHTML = `
    <main></main>
    <div data-kroket-fixed style="padding-right: 1rem"></div>
  `;

  act(() => {
    lock();
    unlock();
  });

  expect(document.body).toMatchSnapshot();
});

it('shouldn\'t lock when there are locked children', () => {
  const { result: { current: [lock] } } = renderHook(() => useLock());

  document.body.innerHTML = `
    <main></main>
    <div data-kroket-fixed style="padding-right: 1rem"></div>
  `;

  act(() => {
    lock();
    lock();
  });

  expect(document.body).toMatchSnapshot();
});
