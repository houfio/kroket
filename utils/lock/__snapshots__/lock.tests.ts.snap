// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should hide overflow on body 1`] = `
<body
  style="overflow: hidden;"
/>
`;

exports[`should preserve padding 1`] = `
<body
  style="overflow: hidden;"
>
  
    
  <main />
  
    
  <div
    data-kroket-fixed=""
    style="padding-right: 1025px;"
  />
  
  
</body>
`;

exports[`should restore fixed padding 1`] = `
<body
  style="overflow: auto;"
>
  
    
  <main />
  
    
  <div
    data-kroket-fixed=""
    style="padding-right: 1px;"
  />
  
  
</body>
`;

exports[`shouldn't lock when there are locked children 1`] = `
<body
  style="overflow: hidden;"
>
  
    
  <main />
  
    
  <div
    data-kroket-fixed=""
    style="padding-right: 1025px;"
  />
  
  
</body>
`;
